src/System.Security.Cryptography.Algorithms/src/System/Security/Cryptography/AesAEAD.Windows.cs

```diff
@@ -0,0 +1,102 @@
+            ReadOnlySpan<byte> nonce,
+            ReadOnlySpan<byte> associatedData,
+            ReadOnlySpan<byte> plaintext,
+            Span<byte> ciphertext,
+            Span<byte> tag)
+        {
+            fixed (byte* plaintextBytes = plaintext)
+            fixed (byte* nonceBytes = nonce)
+            fixed (byte* ciphertextBytes = ciphertext)
+            fixed (byte* tagBytes = tag)
+            fixed (byte* associatedDataBytes = associatedData)
+            {
+                BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO authInfo = BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO.Create();
+                authInfo.pbNonce = nonceBytes;
+                authInfo.cbNonce = nonce.Length;
+                authInfo.pbTag = tagBytes;
+                authInfo.cbTag = tag.Length;
+                authInfo.pbAuthData = associatedDataBytes;
+                authInfo.cbAuthData = associatedData.Length;
+
+                NTSTATUS ntStatus = Cng.Interop.BCryptEncrypt(
+                    keyHandle,
+                    plaintextBytes,
+                    plaintext.Length,
+                    new IntPtr(&authInfo),
+                    null,
+                    0,
+                    ciphertextBytes,
+                    ciphertext.Length,
+                    out int ciphertextBytesWritten,
+                    0);
+
+                if (ntStatus != NTSTATUS.STATUS_SUCCESS)
+                {
+                    throw CreateCryptographicException(ntStatus);
+                }
+            }
+        }
+
+        public static unsafe void Decrypt(
+            SafeAlgorithmHandle algorithm,
+            SafeKeyHandle keyHandle,
+            ReadOnlySpan<byte> nonce,
+            ReadOnlySpan<byte> associatedData,
+            ReadOnlySpan<byte> ciphertext,
+            ReadOnlySpan<byte> tag,
+            Span<byte> plaintext)
+        {
+            fixed (byte* plaintextBytes = plaintext)
+            fixed (byte* nonceBytes = nonce)
+            fixed (byte* ciphertextBytes = ciphertext)
+            fixed (byte* tagBytes = tag)
+            fixed (byte* associatedDataBytes = associatedData)
+            {
+                BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO authInfo = BCRYPT_AUTHENTICATED_CIPHER_MODE_INFO.Create();
+                authInfo.pbNonce = nonceBytes;
+                authInfo.cbNonce = nonce.Length;
+                authInfo.pbTag = tagBytes;
+                authInfo.cbTag = tag.Length;
+                authInfo.pbAuthData = associatedDataBytes;
+                authInfo.cbAuthData = associatedData.Length;
+
+                NTSTATUS ntStatus = Cng.Interop.BCryptDecrypt(
+                    keyHandle,
+                    ciphertextBytes,
+                    ciphertext.Length,
+                    new IntPtr(&authInfo),
+                    null,
+                    0,
+                    plaintextBytes,
+                    plaintext.Length,
+                    out int plaintextBytesWritten,
```

Comment:
I'd add `Debug.Assert(plainTextBytesWritten == ciphertext.Length);` (and similar above), largely as a "yes, we're knowingly not ignoring this", but also just as good detection for bad refactoring.